### ISSUE TITLE
Japanese files and directories can not be deleted

### ISSUE THREAD

**Issue by ciwasa** [nextcloud/server#5360]
### Steps to reproduce
1. create japanese folder "あいうえおか"
2. drop the file ("てすとnc.docx") to the folder created in step 1
3. delete folder "あいうえおか"

### Expected behaviour

Folders and files contained in them are delete

### Actual behaviour

A message informing of the failure to delete the folder is displayed.
The folder does not disappear from the screen and deletion is not completed.

### Server configuration

It also occurs in the demonstration site
https://demo.nextcloud.com/

**Operating system**:

Ubuntu 16.04.2 + docker 

12.0.0-apache as https://hub.docker.com/_/nextcloud/ 

**Web server:**

apache 

**Database:**

sqlite

**PHP version:**

PHP 7.1.5 (cli) (built: Jun  8 2017 22:55:53) ( NTS )
Copyright (c) 1997-2017 The PHP Group
Zend Engine v3.1.0, Copyright (c) 1998-2017 Zend Technologies
    with Zend OPcache v7.1.5, Copyright (c) 1999-2017, by Zend Technologies

**Nextcloud version:** (see Nextcloud admin page)

12.0.0.0

**Updated from an older Nextcloud/ownCloud or fresh install:**

**Where did you install Nextcloud from:**

12.0.0-apache as https://hub.docker.com/_/nextcloud/ 

**Signing status:**
<details>
<summary>Signing status</summary>

```
Login as admin user into your Nextcloud and access 
http://example.com/index.php/settings/integrity/failed 
paste the results here.
```
</details>

**List of activated apps:**

<details>
<summary>App list</summary>

```
windows
 Internet Explorer 11
 Chrome

```
</details>

**Nextcloud configuration:**
<details>
<summary>Config report</summary>

```
default
```
</details>

**Are you using external storage, if yes which one:** local/smb/sftp/...

**Are you using encryption:** yes/no

**Are you using an external user-backend, if yes which one:** LDAP/ActiveDirectory/Webdav/...

#### LDAP configuration (delete this part if not used)
<details>
<summary>LDAP config</summary>

```
With access to your command line run e.g.:
sudo -u www-data php occ ldap:show-config
from within your Nextcloud installation folder

Without access to your command line download the data/owncloud.db to your local
computer or access your SQL server remotely and run the select query:
SELECT * FROM `oc_appconfig` WHERE `appid` = 'user_ldap';


Eventually replace sensitive data as the name/IP-address of your LDAP server or groups.
```
</details>

### Client configuration
**Browser:**

**Operating system:**

#### Web server error log
<details>
<summary>Web server error log</summary>

```
Insert your webserver log here
```
</details>

#### Nextcloud log (data/nextcloud.log)
<details>
<summary>Nextcloud log</summary>

```
Insert your Nextcloud log here
```
</details>

#### Browser log
<details>
<summary>Browser log</summary>

```
screencapture_error.png
```
</details>
[てすとnc.docx](https://github.com/nextcloud/server/files/1070482/nc.docx)

![screencapture_error](https://user-images.githubusercontent.com/18418328/27068829-ed5ebe38-504c-11e7-8895-6765de85e4c6.png)

**Comment by blizzz** [comment]
Anything in the server log?

**Comment by ciwasa** [comment]
Log file of docker image could not be found.
However, logs on the WEB interface were collected.

index.php/settings/admin/logging

> Fatal
> Doctrine\DBAL\Exception\UniqueConstraintViolationException: 
>  An exception occurred while executing 
>   'UPDATE "oc_filecache" SET "storage" = ?, "path" = ?, "path_hash" = ?, "name" = ?, "parent" = ? WHERE "fileid" = ?' with 
>     params [2, "files_trashbin\/files\/\u3042\u3044\u3046\u3048\u304a\u304b.d1497345860", "e594132e1f1db66a92307eb9b8587ddb", 
>      "\u3042\u3044\u3046\u3048\u304a\u304b.d1497345860", 95, 90]: 
>  SQLSTATE[23000]: Integrity constraint violation: 19 UNIQUE constraint failed: 
>   oc_filecache.storage, oc_filecache.path_hash
>

**Comment by blizzz** [comment]
@icewind1991 my guess it that the old entry is somehow not found?

**Comment by ciwasa** [comment]
First of all, no problem has occurred in 11.0.3.
Therefore, I think that it is a problem after version 12.
For simplicity we report on docker (12.0.0-apache) and demonstration site.


I have not found the same case example searched with the following keywords.

> is:issue  "An exception occurred while executing" 
> is:issue  "UNIQUE constraint failed" 

If you know the issue to be helpful, please let me know.
If it can not be resolved, consider changing to 11.0.3.

**Comment by ciwasa** [comment]
Since stacktrace information was collected, it will be posted.

> Fatal    webdav   
>  Doctrine\DBAL\Exception\UniqueConstraintViolationException: An exception occurred while executing 'UPDATE "oc_filecache" SET "storage" = ?, "path" = ?, "path_hash" = ?, "name" = ?, "parent" = ? WHERE "fileid" = ?' with params [2, "files_trashbin\/files\/\u3042\u3044\u3046\u3048\u304a\u304b.d1497421599", "10a75eeca6a49359178d3748a81a2016", "\u3042\u3044\u3046\u3048\u304a\u304b.d1497421599", 106, 101]: SQLSTATE[23000]: Integrity constraint violation: 19 UNIQUE constraint failed: oc_filecache.storage, oc_filecache.path_hash
> 
> /var/www/html/3rdparty/doctrine/dbal/lib/Doctrine/DBAL/DBALException.php - line 128: Doctrine\DBAL\Driver\AbstractSQLiteDriver->convertException('An exception oc...', Object(Doctrine\DBAL\Driver\PDOException))
> /var/www/html/3rdparty/doctrine/dbal/lib/Doctrine/DBAL/Connection.php - line 855: Doctrine\DBAL\DBALException driverExceptionDuringQuery(Object(Doctrine\DBAL\Driver\PDOSqlite\Driver), Object(Doctrine\DBAL\Driver\PDOException), 'UPDATE "oc_file...', Array)
> /var/www/html/lib/private/DB/Connection.php - line 192: Doctrine\DBAL\Connection->executeQuery('UPDATE "oc_file...', Array, Array, NULL)
> /var/www/html/lib/private/Files/Cache/Cache.php - line 553: OC\DB\Connection->executeQuery('UPDATE "oc_file...', Array)
> /var/www/html/lib/private/Files/Cache/Cache.php - line 484: OC\Files\Cache\Cache->moveFromCache(Object(OC\Files\Cache\HomeCache), 'files/\xE3\x81\x82\xE3\x81\x84\xE3\x81\x86...', 'files_trashbin/...')
> /var/www/html/lib/private/Files/Cache/Updater.php - line 196: OC\Files\Cache\Cache->move('files/\xE3\x81\x82\xE3\x81\x84\xE3\x81\x86...', 'files_trashbin/...')
> /var/www/html/apps/files_trashbin/lib/Trashbin.php - line 267: OC\Files\Cache\Updater->renameFromStorage(Object(OCA\Files_Trashbin\Storage), 'files/\xE3\x81\x82\xE3\x81\x84\xE3\x81\x86...', 'files_trashbin/...')
> /var/www/html/apps/files_trashbin/lib/Storage.php - line 247: OCA\Files_Trashbin\Trashbin move2trash('\xE3\x81\x82\xE3\x81\x84\xE3\x81\x86\xE3\x81\x88\xE3\x81\x8A...', false)
> /var/www/html/apps/files_trashbin/lib/Storage.php - line 189: OCA\Files_Trashbin\Storage->doDelete('files/\xE3\x81\x82\xE3\x81\x84\xE3\x81\x86...', 'rmdir')
> /var/www/html/lib/private/Files/View.php - line 1132: OCA\Files_Trashbin\Storage->rmdir('files/\xE3\x81\x82\xE3\x81\x84\xE3\x81\x86...')
> /var/www/html/lib/private/Files/View.php - line 348: OC\Files\View->basicOperation('rmdir', '/\xE3\x81\x82\xE3\x81\x84\xE3\x81\x86\xE3\x81\x88\xE3\x81...', Array)
> /var/www/html/apps/dav/lib/Connector/Sabre/Directory.php - line 303: OC\Files\View->rmdir('/\xE3\x81\x82\xE3\x81\x84\xE3\x81\x86\xE3\x81\x88\xE3\x81...')
> /var/www/html/3rdparty/sabre/dav/lib/DAV/Tree.php - line 179: OCA\DAV\Connector\Sabre\Directory->delete()
> /var/www/html/3rdparty/sabre/dav/lib/DAV/CorePlugin.php - line 287: Sabre\DAV\Tree->delete('\xE3\x81\x82\xE3\x81\x84\xE3\x81\x86\xE3\x81\x88\xE3\x81\x8A...')
> [internal function] Sabre\DAV\CorePlugin->httpDelete(Object(Sabre\HTTP\Request), Object(Sabre\HTTP\Response))
> /var/www/html/3rdparty/sabre/event/lib/EventEmitterTrait.php - line 105: call_user_func_array(Array, Array)
> /var/www/html/3rdparty/sabre/dav/lib/DAV/Server.php - line 479: Sabre\Event\EventEmitter->emit('method DELETE', Array)
> /var/www/html/3rdparty/sabre/dav/lib/DAV/Server.php - line 254: Sabre\DAV\Server->invokeMethod(Object(Sabre\HTTP\Request), Object(Sabre\HTTP\Response))
> /var/www/html/apps/dav/appinfo/v1/webdav.php - line 71: Sabre\DAV\Server->exec()
> /var/www/html/remote.php - line 162: require_once('/var/www/html/a...')
> {main}

**Comment by MorrisJobke** [comment]
UniqueConstraintViolationException <- this was fixed in 14.0.4, 13.0.8 and 12.0.13.
